packages/node/src/policy-gen/policy-gen-compartment-class.js

```diff
@@ -163,48 +173,63 @@ const makeGetOverriddenResourceNames = (compartmentMap, resources) => {
         return
       }
 
-      /* c8 ignore next */
-      if (compartmentMap.entry.compartment === compartmentName) {
-        // This also "should never happen", as a foreign compartment should
-        // not be importing the entry compartment. Also, the entry
-        // compartment cannot have a canonical name, and we need a canonical
-        // name to apply policy
-        throw new ReferenceError(
-          `Unexpected entry compartment encountered in ${moduleDescriptorCompartment.label}; this is a bug`
-        )
```

Comment:
why is this gone?

Reply:
I was hitting it when dogfooding 

- [ ] double-check that's really what's happening here.

Reply:
(just gut feeling) might be that the mechanism for breaking the cycle when something tries to import from root compartment is depending accidentally on it being a trusted root.

Reply:
will look at this in #1576